cloud/amazon/ec2_asg_facts.py

```diff
@@ -0,0 +1,410 @@
+                "auto_scaling_group_name": "public-webapp-production",
+                "availability_zones": ["us-west-2c", "us-west-2b", "us-west-2a"],
+                "created_time": "2016-02-02T23:28:42.481000+00:00",
+                "default_cooldown": 300,
+                "desired_capacity": 2,
+                "enabled_metrics": [],
+                "health_check_grace_period": 300,
+                "health_check_type": "ELB",
+                "instances":
+                [
+                    {
+                        "availability_zone": "us-west-2c",
+                        "health_status": "Healthy",
+                        "instance_id": "i-047a12cb",
+                        "launch_configuration_name": "public-webapp-production-1",
+                        "lifecycle_state": "InService",
+                        "protected_from_scale_in": false
+                    },
+                    {
+                        "availability_zone": "us-west-2a",
+                        "health_status": "Healthy",
+                        "instance_id": "i-7a29df2c",
+                        "launch_configuration_name": "public-webapp-production-1",
+                        "lifecycle_state": "InService",
+                        "protected_from_scale_in": false
+                    }
+                ],
+                "launch_configuration_name": "public-webapp-production-1",
+                "load_balancer_names": ["public-webapp-production-lb"],
+                "max_size": 4,
+                "min_size": 2,
+                "new_instances_protected_from_scale_in": false,
+                "placement_group": None,
+                "status": None,
+                "suspended_processes": [],
+                "tags":
+                [
+                    {
+                        "key": "Name",
+                        "propagate_at_launch": true,
+                        "resource_id": "public-webapp-production",
+                        "resource_type": "auto-scaling-group",
+                        "value": "public-webapp-production"
+                    },
+                    {
+                        "key": "env",
+                        "propagate_at_launch": true,
+                        "resource_id": "public-webapp-production",
+                        "resource_type": "auto-scaling-group",
+                        "value": "production"
+                    }
+                ],
+                "termination_policies":
+                [
+                    "Default"
+                ],
+                "vpc_zone_identifier":
+                [
+                    "subnet-a1b1c1d1",
+                    "subnet-a2b2c2d2",
+                    "subnet-a3b3c3d3"
+                ]
+            }
+        ]
+    """
+    asgs = conn.describe_auto_scaling_groups()['AutoScalingGroups']
+    matched_asgs = []
```

Comment:
In this case, the user can specify either, both or neither of the `name` and `tags` arguments. The name can also be a partial match and the API call doesn't seem to support wildcards. There are also no other filter options for the API call beyond specifying a list of *exact* names.